src/hooks/useCustomClient.tsx

```diff
@@ -1,42 +1,39 @@
 import { useMemo } from 'react';
-import { PublicClient, useNetwork } from 'wagmi';
-import { goerli, mainnet, optimism, arbitrum, polygon } from 'wagmi/chains';
+import { PublicClient, useAccount } from 'wagmi';
+import { goerli, mainnet, optimism, arbitrum, polygon, sepolia } from 'wagmi/chains';
 import { Chain as WagmiChain, createPublicClient, custom, fallback, http } from 'viem';
 import 'viem/window';
 
 import { getConfig } from '~/config';
-import { useStateContext } from '~/hooks';
+import { Chains } from '~/types';
 
-export const useCustomClient = () => {
-  const { userAddress } = useStateContext();
-  const { TEST_MODE, DEFAULT_CHAIN, availableChains, ALCHEMY_KEY } = getConfig();
-  const { chain } = useNetwork();
+const { TEST_MODE, DEFAULT_CHAIN, availableChains, ALCHEMY_KEY } = getConfig();
 
-  const customClient = useMemo(() => {
-    const getAlchemyProvider = (chainId: number) => {
-      const apiUrl = availableChains[chainId]?.alchemyUrl || availableChains[DEFAULT_CHAIN]?.alchemyUrl;
-      return `${apiUrl}/${ALCHEMY_KEY}`;
-    };
+const getAlchemyProvider = (chainId: number, availableChains: Chains) => {
+  const apiUrl = availableChains[chainId]?.alchemyUrl || availableChains[DEFAULT_CHAIN]?.alchemyUrl;
+  return `${apiUrl}/${ALCHEMY_KEY}`;
+};
 
-    const chainId = TEST_MODE ? DEFAULT_CHAIN : chain?.id || DEFAULT_CHAIN;
+export const useCustomClient = (chainId: number) => {
+  const { address: userAddress } = useAccount();
 
-    const alchemy = http(getAlchemyProvider(chainId), { batch: true });
+  const customClient = useMemo(() => {
+    const alchemy = http(getAlchemyProvider(chainId || DEFAULT_CHAIN, availableChains), { batch: true });
     const customTransport = custom(window.ethereum!);
 
     // this allows to use the connected wallet rpc provider
     const transportFallback = fallback(userAddress ? [customTransport, alchemy] : [alchemy]);
 
     const selectedChain: WagmiChain =
-      [goerli, mainnet, optimism, arbitrum, polygon].find((chain) => chain?.id === chainId) || goerli;
-
+      [goerli, mainnet, optimism, arbitrum, polygon, sepolia].find((chain) => chain?.id === chainId) || sepolia;
```

Comment:
create constant AVAILABLE_CHAINS in some place, and replace it in all the app since it is used in multiple places

```suggestion
      [AVAILABLE_CHAINS].find((chain) => chain?.id === chainId) || sepolia;
```

Reply:
done https://github.com/defi-wonderland/xkeeper-app/pull/28/commits/8e538af19fb5716d813c2940514b502dfd2a2b71